### ISSUE TITLE
Existing "stack ghc compiler version not found"

### ISSUE THREAD

**Issue by andreasabel** [juhp/stack-clean-old#14]
I am getting "stack ghc compiler version 9.2.5 not found" even though this version looks like it is installed.

```shell-session
$ stack-clean-old remove -C 9.2.5
stack ghc compiler version 9.2.5 not found

$ stack-clean-old list -C
...
ghc-9.2.5
ghc-9.4.3.temp
ghc-9.4.4

~/.stack/programs/x86_64-osx$ ls -al
...
drwxr-xr-x   6 abel  staff        192 Nov 17 10:01 ghc-9.2.5
-rw-------   1 abel  staff          9 Nov 17 10:01 ghc-9.2.5.installed
-rw-r--r--   1 abel  staff  278203264 Nov 17 09:58 ghc-9.2.5.tar.bz2
-rw-r--r--   1 abel  staff  295581084 Nov 19 16:30 ghc-9.4.3.tar.bz2
drwxr-xr-x   3 abel  staff         96 Jan 15 14:39 ghc-9.4.3.temp
drwxr-xr-x   5 abel  staff        160 Jan 14 21:54 ghc-9.4.4
-rw-------   1 abel  staff          9 Jan 14 21:55 ghc-9.4.4.installed
-rw-r--r--   1 abel  staff  295620185 Jan 14 21:51 ghc-9.4.4.tar.bz2
```

**Comment by juhp** [comment]
Hmm, thanks for the report.

I think this needs a bit of debugging... not obvious to me either what is going on...

**Comment by andreasabel** [comment]
Same for 9.4.5 now:
```
.stack/programs/x86_64-osx$ l
total 5827176
drwxr-xr-x  38 abel  staff       1216 Aug 21 14:33 .
drwxr-xr-x   3 abel  staff         96 Jan  2  2019 ..
drwxr-xr-x   5 abel  staff        160 Dec 16  2022 ghc-7.10.3
-rw-------   1 abel  staff          9 Dec 16  2022 ghc-7.10.3.installed
-rw-r--r--   1 abel  staff  147607506 Dec 16  2022 ghc-7.10.3.tar.bz2
drwxr-xr-x   5 abel  staff        160 Jan 16  2021 ghc-8.0.2
-rw-------   1 abel  staff          9 Jan 16  2021 ghc-8.0.2.installed
-rw-r--r--   1 abel  staff  186822039 Jan 16  2021 ghc-8.0.2.tar.bz2
drwxr-xr-x   5 abel  staff        160 Sep 19  2021 ghc-8.10.7
-rw-------   1 abel  staff          9 Sep 19  2021 ghc-8.10.7.installed
-rw-r--r--   1 abel  staff  274034358 Sep 19  2021 ghc-8.10.7.tar.bz2
drwxr-xr-x   5 abel  staff        160 Oct 26  2020 ghc-8.2.2
-rw-------   1 abel  staff          9 Oct 26  2020 ghc-8.2.2.installed
-rw-r--r--   1 abel  staff  192467920 Oct 26  2020 ghc-8.2.2.tar.bz2
drwxr-xr-x   5 abel  staff        160 Mar 15  2019 ghc-8.4.4
-rw-r--r--   1 abel  staff          9 Mar 15  2019 ghc-8.4.4.installed
-rw-r--r--   1 abel  staff  230734167 Mar 15  2019 ghc-8.4.4.tar.bz2
drwxr-xr-x   5 abel  staff        160 Oct 26  2020 ghc-8.6.5
-rw-------   1 abel  staff          9 Oct 26  2020 ghc-8.6.5.installed
-rw-r--r--   1 abel  staff  226259652 Oct 26  2020 ghc-8.6.5.tar.bz2
drwxr-xr-x   5 abel  staff        160 Dec 25  2020 ghc-8.8.4
-rw-------   1 abel  staff          9 Dec 25  2020 ghc-8.8.4.installed
-rw-r--r--   1 abel  staff  272402324 Dec 25  2020 ghc-8.8.4.tar.bz2
drwxr-xr-x   6 abel  staff        192 Jan 31  2022 ghc-9.0.2
-rw-------   1 abel  staff          9 Jan 31  2022 ghc-9.0.2.installed
-rw-r--r--   1 abel  staff  264948596 Jan 31  2022 ghc-9.0.2.tar.bz2
drwxr-xr-x   6 abel  staff        192 Jun 10 12:27 ghc-9.2.8
-rw-------   1 abel  staff          9 Jun 10 12:28 ghc-9.2.8.installed
-rw-r--r--   1 abel  staff  280241113 Jun 10 12:26 ghc-9.2.8.tar.bz2
drwxr-xr-x   5 abel  staff        160 Jun 10 12:15 ghc-9.4.5
-rw-------   1 abel  staff          9 Jun 10 12:18 ghc-9.4.5.installed
-rw-r--r--   1 abel  staff  296264786 Jun 10 12:10 ghc-9.4.5.tar.bz2
drwxr-xr-x   5 abel  staff        160 Aug 21 14:31 ghc-9.4.6
-rw-------   1 abel  staff          9 Aug 21 14:33 ghc-9.4.6.installed
-rw-r--r--   1 abel  staff  296491411 Aug 21 14:27 ghc-9.4.6.tar.bz2
drwxr-xr-x   5 abel  staff        160 Jun 22 21:24 ghc-9.6.2
-rw-------   1 abel  staff          9 Jun 22 21:26 ghc-9.6.2.installed
-rw-r--r--   1 abel  staff  309039548 Jun 22 21:21 ghc-9.6.2.tar.bz2

$ stack-clean-old list -C
ghc-7.10.3
ghc-8.0.2
ghc-8.2.2
ghc-8.4.4
ghc-8.6.5
ghc-8.8.4
ghc-8.10.7
ghc-9.0.2
ghc-9.2.8
ghc-9.4.5
ghc-9.4.6
ghc-9.6.2

$ stack-clean-old remove 9.4.5 -C
stack ghc compiler version 9.4.5 not found
```

**Comment by andreasabel** [comment]
https://github.com/juhp/stack-clean-old/blob/18bb9ef6c7e2214ccc23c1acf1f3fc622022d469/src/GHC.hs#L33-L42
The last line looks wrong.  (Why the `"*-"`?)

I think I fixed the bug in PR #16.

**Comment by juhp** [comment]
Thanks @andreasabel for this.

If it doesn't look too bad, I may push out a release now.
We can still iterate on improving this more later if needed.

**Comment by juhp** [comment]
Released in [0.4.7](https://hackage.haskell.org/package/stack-clean-old-0.4.7)
